### PR TITLE
Adding Multiple rosbridge Connections and ROS2 Support

### PR DESCRIPTION
**Pull request by tsender**
This PR adds two big improvements to the plugin.

1. One addition is the ability to connect to multiple rosbridge servers. Most users probably won't need this feature, but it's a feature that may come in handy. For example, if someone has large data sensors (camera or lidar), then sending all of this data across single a rosbridge connection can create a bottleneck for all the other important messages that need to get through. Connecting to multiple servers can allow each sensor to use its own rosbridge server. Unfortunately, this speedup will only be noticed if using "tcp", since Unreal's websocket module only uses a single thread to handle all websocket connections.
   - I did rearrange and clean up a fair amount of code in the ROSIntegrationGameInstance.cpp file while keeping the same sequence of commands. I added arrays to specify the rosbridge server IPs and ports. For legacy purposes, to avoid existing users from having any compile errors, the `ROSIntegrationCore` variable is still a valid pointer.
2. The second addition is the ability to connect to a rosbridge server running either ROS1 or ROS2. The easiest way I could find to implement this without modifying too many things was to create a variable `ROS_VERSION` that can be accessed by numerous classes with the use of the "extern" keyword. I also modified the existing converters to automatically handle adding/extracting the appropriate timestamp fields.

To the best of my knowledge, no one should encounter any compile errors due to the addition of these changes. I kept a few things around for legacy reasons and made sure things still work as expected. I have tested these changes with UE4.26 on Windows and everything works as I would expect it to.

I also updated the README to account for all these changes. 

Since a lot of things were changed, feel free to take your time to look through the changes or make any suggestions before merging.

### CODE REVIEW

**Comment by tsender** [comment]
@Sanic, have you had any time to take a look at this yet?

**Comment by Sanic** [comment]
Hi @tsender 
Not yet unfortunately. But it's still on my list 🙂

File: README.md [hunk 1]
```diff
@@ -94,10 +96,16 @@ Please note that this list is a tracker of which UE4 versions have been previous
 
     ![Create a new GameInstance](Documentation/ue4-setup-01.png)
 
-  * Open your new C++ class / Blueprint object and change the values of `ROSBridgeSeverHost` and `ROSBridgeServerPort`
-
-    ![Change host and port to match your server](Documentation/ue4-setup-02.png)
-
+  * Open your new C++ class / Blueprint object and configure the rosbridge connection properties:
+    * `ROSBridgeServerProtocol`: The communication protocol to use with rosbridge. Options are "tcp" or "ws".
+    * `ROSBridgeSeverHosts`: A list of rosbridge server IP addresses. Each element pairs with the corresponding element in `ROSBridgeSeverPorts`.
```
Comment (Sanic) [inline README.md:101]: Small typo: 'Sever' instead of 'Server'. This also happened with the Ports.

**Review by Sanic — COMMENTED** [review]
What a great addition! Thanks a lot.
I only marked a requested change for the README. 
Besides that, the PR looks good 👍

**Comment by tsender** [comment]
Thanks. I just fixed the typos.

**Comment by Sanic** [comment]
Thanks a lot for all the changes! 🎉

**Pull request merged**